pkg/network/base_connection.go

```diff
@@ -154,45 +173,20 @@ func (t *baseTcpConnection) consumeConnection(conn net.Conn) {
 	wg.Wait()
 
 	t.logger.Debugf("Stopped consuming connection with local %s and remote %s", conn.LocalAddr().String(), conn.RemoteAddr().String())
-	t.connMutex.RLock()
-	err := t.conn.Close()
-	t.connMutex.RUnlock()
-	if err != nil && !isEOF(err) && !isUseOfClosedConnection(err) {
-		t.logger.Warnf("Error while closing the previous connection: %s", err)
-	}
 }
 
-func (t *baseTcpConnection) tryPropagateError(err error) {
-	select {
-	case <-t.ctx.Done():
-		return
-	default:
-		t.errors <- err
-	}
+func (t *baseTcpConnection) pushError(err error) {
+	t.errors <- err
 }
```

Comment:
Personally, I'd ditch this function and the following. They seem kinda useless and make you do more work to find out what's actually happening.